src/mlpack/tests/mlpack_test.cpp

```diff
@@ -47,6 +106,22 @@ struct GlobalFixture
       mlpack::Log::Info.ignoreInput = true;
       mlpack::Log::Warn.ignoreInput = true;
     #endif
+
+    for (int i = 0; i < boost::unit_test::framework::master_test_suite().argc;
+        i++)
+    {
+      std::string argument(
+          boost::unit_test::framework::master_test_suite().argv[i]);
+
+      // Print Boost test hierarchy.
+      if (argument == "tests")
```

Comment:
I think I found a solution, for newer boost versions (>= 103600) there is already an option to print the test cases `` --list_content``. We can just replicate the behaviour and since we can also use ``--list_content`` for older versions we don't have the problem with the issue I pointed out earlier.